RPerformanceTracking/Private/_RPTTracker.m

```diff
@@ -103,7 +103,12 @@ - (void)end:(uint_fast64_t)trackingIdentifier statusCode:(NSInteger)statusCode
     if (trackingIdentifier)
     {
         _RPTMeasurement *measurement = [_ringBuffer measurementWithTrackingIdentifier:trackingIdentifier];
-        measurement.statusCode = statusCode;
+
+        // The status code of the measurement can be reset to 0 by the -(void)end: method, so we need to check the status code before setting it to the measurement.
+        if (statusCode != 0)
+        {
+            measurement.statusCode = statusCode;
+        }
```

Comment:
As discussed, instead of this workaround I think it would be better to replace this method with an `updateMeasurement:statusCode` method that is called from the decidePolicyForNavigationResponse swizzle